operations/helm/charts/mimir-distributed/templates/_helpers.tpl

```diff
@@ -74,6 +74,18 @@ Create the name of the service account
 {{- end -}}
 {{- end -}}
 
+{{/*
+Create the name of the ruler service account
+*/}}
+{{- define "mimir.ruler.serviceAccountName" -}}
+{{- if .Values.ruler.serviceAccount.create -}}
```

Comment:
the behaviour with the default SA is that people might opt to not have the chart create it, but still have the pods associate with the name there. Should we take the same approach? 

Instead of `if .Values.ruler.serviceAccount.create` we can do `if .Values.ruler.serviceAccount.name`. The default of `null` for the name in values.yaml will just end up using general SA.

I know I'm suggesting to change this line for the second time, sorry about that